File: src/local_ai_agent/router/classifier.py
from __future__ import annotations


def normalize_input(raw_input: str) -> str:
    return " ".join(raw_input.strip().split())


def parse_command_shape(normalized_input: str) -> tuple[str, list[str]]:
    parts = normalized_input.split()
    if not parts:
        return "", []
    return parts[0], parts[1:]

File: crates/router-core/src/classifier.rs
pub fn normalize_input(raw_input: &str) -> String {
    raw_input.split_whitespace().collect::<Vec<_>>().join(" ")
}

pub fn parse_command_shape(normalized_input: &str) -> (String, Vec<String>) {
    let mut parts = normalized_input.split_whitespace();
    let Some(command) = parts.next() else {
        return (String::new(), Vec::new());
    };

    (
        command.to_string(),
        parts.map(ToOwned::to_owned).collect::<Vec<_>>(),
    )
}
